Query.js

```diff
@@ -0,0 +1,172 @@
+/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "_" }] */
+/* globals IsNumeric_, wrapValue_ */
+
+/**
+ * An object that acts as a Query to be a structured query.
+ *
+ * @class
+ * @private
+ * @see {@link https://firebase.google.com/docs/firestore/reference/rest/v1beta1/StructuredQuery Firestore Structured Query}
+ * @param {string} from the base collection to query
+ * @param {string} callback the function that is executed with the internally compiled query
+ */
+var FirestoreQuery_ = function (from, callback) {
+  const this_ = this
+
+  // @see {@link https://firebase.google.com/docs/firestore/reference/rest/v1beta1/StructuredQuery#Operator_1 FieldFilter Operator}
+  const fieldOps = {
+    '==': 'EQUAL',
+    '===': 'EQUAL',
+    '<': 'LESS_THAN',
+    '<=': 'LESS_THAN_OR_EQUAL',
+    '>': 'GREATER_THAN',
+    '>=': 'GREATER_THAN_OR_EQUAL'
+  }
+
+  // @see {@link https://firebase.google.com/docs/firestore/reference/rest/v1beta1/StructuredQuery#Operator_2 FieldFilter Operator}
+  const unaryOps = {
+    'nan': 'IS_NAN',
+    'null': 'IS_NULL'
+  }
+
+  // @see {@link https://firebase.google.com/docs/firestore/reference/rest/v1beta1/StructuredQuery#FieldReference Field Reference}
+  const fieldRef = function (field) {
+    return {fieldPath: field}
+  }
+  const filter = function (field, operator, value) {
+    // @see {@link https://firebase.google.com/docs/firestore/reference/rest/v1beta1/StructuredQuery#FieldFilter Field Filter}
+    if (operator in fieldOps) {
+      return {
+        fieldFilter: {
+          field: fieldRef(field),
+          op: fieldOps[operator],
+          value: wrapValue_(value)
+        }
+      }
+    }
+
+    // @see {@link https://firebase.google.com/docs/firestore/reference/rest/v1beta1/StructuredQuery#UnaryFilter Unary Filter}
+    if (operator.toLowerCase() in unaryOps) {
+      return {
+        unaryFilter: {
+          field: fieldRef(field),
+          op: unaryOps[operator]
+        }
+      }
+    }
+    throw new Error('Invalid Operator given ' + operator)
+  }
+
+  const query = {
+    from: from.map(function (collection) {
```

Comment:
This line confused me for a while since I was thinking `from` was a string. It's an array, though, right? Could you update the class-level comment to say this parameter is an array?

Reply:
`from` gets transformed into an array. I guess I overlooked that JSDoc comment. Thanks!